### ISSUE TITLE
Support for Managed Identities  in azurerm_automation_account resource

### ISSUE THREAD

**Issue by ivanov-aleksander** [hashicorp/terraform-provider-azurerm#11503]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s)

* azurerm_automation_account

### Potential Terraform Configuration

```hcl
# Identity can be configured as for azurerm_linux_virtual_machine resource
# A identity block supports the following:
#   type - (Required) The type of Managed Identity which should be assigned to the Azure Automation Account. Possible values are SystemAssigned, UserAssigned
#   identity_ids - (is required when type is set to UserAssigned) A list of User Managed Identity ID's which should be assigned to the Azure Automation Account.


resource "azurerm_automation_account" "example" {
  name                = "automationAccount1"
  ...
  identity = {
     type = "SystemAssigned"
  }
}
```

### References

* [Identity argument reference for azurerm_linux_virtual_machine resource](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine#identity)
* [Azure Automation account authentication by Managed Identities ](https://docs.microsoft.com/en-us/azure/automation/enable-managed-identity-for-automation)

**Comment by ivanov-aleksander** [comment]
I hope this feature will be implemented

**Comment by zic04** [comment]
Hello, no update on the subject ?

**Comment by swupdiedoowap** [comment]
We also would really like this feature.

**Comment by brakelee** [comment]
We also want to use this feature.

**Comment by lpmagness1** [comment]
Is there any update on this feature? We need this. Any good workarounds?

**Comment by kevinaschmidt** [comment]
I've seen that this functionality is available on virtual machines, Any reason it isn't also available here? Could we get a roadmap of when we can get this?

**Comment by swupdiedoowap** [comment]
Apperently the azure sdk for go is not yet updated with the identity property. So that might explain why it is not there yet. 
https://github.com/Azure/azure-sdk-for-go/issues/15020

**Comment by alxgda** [comment]
Microsoft really need to gives a huge boost to Automation Account regarding IaC features. That, and no ability to specify a PowerShell version are a big turn off.

**Comment by lpmagness1** [comment]
> Apperently the azure sdk for go is not yet updated with the identity property. So that might explain why it is not there yet. [Azure/azure-sdk-for-go#15020](https://github.com/Azure/azure-sdk-for-go/issues/15020)

I believe they just completed a PR for this recently: https://github.com/Azure/azure-sdk-for-go/pull/16837

**Comment by manssa01** [comment]
Hi all,

I am now more confused than before!!
Is there a new feature to support managed identity for azurerm_automation_account or no?

Thanks and best regards,

**Comment by nickwb** [comment]
Hi all,

According to the docs, it looks like an `identity` block is supported for `azurerm_automation_account`, but there is no way to retrieve the object id of a system assigned identity?

Kind Regards

**Comment by dsfrederic** [comment]
@nickwb yes there is! 

```HCL
output "identity" {
  value = azurerm_automation_account.aut.identity[0].principal_id
}
```

**Comment by nickwb** [comment]
> @nickwb yes there is!
> 
> ```terraform
> output "identity" {
>   value = azurerm_automation_account.aut.identity[0].principal_id
> }
> ```

Oh, I see, thanks for the hint @dsfrederic.

I was correct, at least, in my assertion that it is missing from the latest documentation:
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/automation_account

**Comment by aristosvo** [comment]
Fixed in [#16007](https://github.com/hashicorp/terraform-provider-azurerm/pull/16007), I'll close this issue because of that.

**Comment by KezHalls** [comment]
This feature does not work for me... can this issue be reopened or do I need to open another.
Error:  Blocks of type "identity" are not expected here.  tried with "=" and without
```
identity = {
        type         = "UserAssigned"
        identity_ids = ["ex@somewhere.com"]
```
